dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/PoolTaskRestRepository.java

```diff
@@ -0,0 +1,137 @@
+import org.dspace.app.rest.Parameter;
+import org.dspace.app.rest.SearchRestMethod;
+import org.dspace.app.rest.converter.PoolTaskConverter;
+import org.dspace.app.rest.model.PoolTaskRest;
+import org.dspace.app.rest.model.hateoas.PoolTaskResource;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.event.Event;
+import org.dspace.workflow.WorkflowException;
+import org.dspace.xmlworkflow.WorkflowConfigurationException;
+import org.dspace.xmlworkflow.factory.XmlWorkflowServiceFactory;
+import org.dspace.xmlworkflow.service.WorkflowRequirementsService;
+import org.dspace.xmlworkflow.service.XmlWorkflowService;
+import org.dspace.xmlworkflow.state.Step;
+import org.dspace.xmlworkflow.state.Workflow;
+import org.dspace.xmlworkflow.state.actions.WorkflowActionConfig;
+import org.dspace.xmlworkflow.storedcomponents.PoolTask;
+import org.dspace.xmlworkflow.storedcomponents.service.PoolTaskService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the repository responsible to manage PooledTask Rest object
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ */
+
+@Component(PoolTaskRest.CATEGORY + "." + PoolTaskRest.NAME)
+public class PoolTaskRestRepository extends DSpaceRestRepository<PoolTaskRest, Integer> {
+
+    private static final Logger log = Logger.getLogger(PoolTaskRestRepository.class);
+
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    EPersonService epersonService;
+
+    @Autowired
+    PoolTaskService poolTaskService;
+
+    @Autowired
+    PoolTaskConverter converter;
+
+    @Autowired
+    XmlWorkflowService workflowService;
+
+    @Autowired
+    WorkflowRequirementsService workflowRequirementsService;
+
+    @Override
+    public PoolTaskRest findOne(Context context, Integer id) {
+        PoolTask task = null;
+        try {
+            task = poolTaskService.find(context, id);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        if (task == null) {
+            return null;
+        }
+        return converter.fromModel(task);
+    }
+
+    @SearchRestMethod(name = "findByUser")
+    public Page<PoolTaskRest> findByUser(@Parameter(value = "uuid") UUID userID, Pageable pageable) {
+        List<PoolTask> tasks = null;
+        try {
+            Context context = obtainContext();
+            EPerson ep = epersonService.find(context, userID);
+            tasks = poolTaskService.findByEperson(context, ep);
+        } catch (SQLException | AuthorizeException | IOException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
Again, a more useful error message. `"Could not find Pooled Tasks for user with ID" + userID`

Reply:
see the other comments. AuthrorizeException has been properly wrapped now

Reply:
Withdrawing my request & resolving this conversation.